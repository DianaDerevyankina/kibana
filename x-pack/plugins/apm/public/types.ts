/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

export interface ApmPluginSetup {
  getGreeting: () => string;
}
// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface ApmPluginStart {}

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface AppPluginStartDependencies {}
