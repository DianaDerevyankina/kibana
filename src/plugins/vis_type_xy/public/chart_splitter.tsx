/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * and the Server Side Public License, v 1; you may not use this file except in
 * compliance with, at your election, the Elastic License or the Server Side
 * Public License, v 1.
 */

import React from 'react';
import { Accessor, AccessorFn, GroupBy, GroupBySort, SmallMultiples } from '@elastic/charts';

interface ChartSplitterProps {
  splitColumnAccessor?: Accessor | AccessorFn;
  splitRowAccessor?: Accessor | AccessorFn;
  sort?: GroupBySort;
}

export const ChartSplitter = ({
  splitColumnAccessor,
  splitRowAccessor,
  sort,
}: ChartSplitterProps) =>
  splitColumnAccessor || splitRowAccessor ? (
    <>
      <GroupBy
        id="__chart_splitter__"
        by={(spec, datum) => {
          const splitTypeAccessor = splitColumnAccessor || splitRowAccessor;
          if (splitTypeAccessor) {
            return typeof splitTypeAccessor === 'function'
              ? splitTypeAccessor(datum)
              : datum[splitTypeAccessor];
          }
          return spec.id;
        }}
        sort={sort || 'dataIndex'}
      />
      <SmallMultiples
        splitVertically={splitRowAccessor ? '__chart_splitter__' : undefined}
        splitHorizontally={splitColumnAccessor ? '__chart_splitter__' : undefined}
      />
    </>
  ) : null;
