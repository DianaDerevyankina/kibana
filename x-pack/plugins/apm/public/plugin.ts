/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */
import { CoreSetup, CoreStart, Plugin } from 'src/core/public';
import { ApmPluginSetup, ApmPluginStart } from './types';

export class ApmPlugin implements Plugin<ApmPluginSetup, ApmPluginStart> {
  public setup(core: CoreSetup): ApmPluginSetup {
    return {};
  }

  public start(core: CoreStart): ApmPluginStart {
    return {};
  }

  public stop() {}
}
