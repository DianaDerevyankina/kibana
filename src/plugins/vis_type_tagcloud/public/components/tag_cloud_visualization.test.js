/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import 'jest-canvas-mock';

import { ImageComparator } from 'test_utils/image_comparator';
import { ExprVis } from '../../../visualizations/public/expressions/vis';

// Replace with mock when converting to jest tests
import { seedColors } from '../../../charts/public/services/colors/seed_colors';
import { BaseVisType } from '../../../visualizations/public/vis_types/base_vis_type';
import { createTagCloudVisTypeDefinition } from '../tag_cloud_type';
import { createTagCloudVisualization } from './tag_cloud_visualization';
import { setFormatService } from '../services';
import { fieldFormatsServiceMock } from '../../../data/public/field_formats/mocks';
import { setHTMLElementOffset, setSVGElementGetBBox } from '../../../../test_utils/public/helpers';

describe('TagCloudVisualizationTest', function () {
  setSVGElementGetBBox(512, 512);

  let domNode;
  let vis;
  let imageComparator;

  const dummyTableGroup = {
    columns: [
      {
        id: 'col-0',
        title: 'geo.dest: Descending',
      },
      {
        id: 'col-1',
        title: 'Count',
      },
    ],
    rows: [
      { 'col-0': 'CN', 'col-1': 26 },
      { 'col-0': 'IN', 'col-1': 17 },
      { 'col-0': 'US', 'col-1': 6 },
      { 'col-0': 'DE', 'col-1': 4 },
      { 'col-0': 'BR', 'col-1': 3 },
    ],
  };
  const TagCloudVisualization = createTagCloudVisualization({
    colors: {
      seedColors,
    },
  });

  beforeAll(() => {
    setFormatService(fieldFormatsServiceMock.createStartContract());
  });

  describe('TagCloudVisualization - basics', function () {
    beforeEach(async function () {
      const visType = new BaseVisType(createTagCloudVisTypeDefinition({ colors: seedColors }));
      setupDOM(512, 512);
      imageComparator = new ImageComparator();
      vis = new ExprVis({
        type: visType,
        params: {
          bucket: { accessor: 0, format: {} },
          metric: { accessor: 0, format: {} },
        },
        data: {},
      });
    });

    afterEach(function () {
      teardownDOM();
      imageComparator.destroy();
    });

    test('simple draw', async function () {
      const tagcloudVisualization = new TagCloudVisualization(domNode, vis);

      await tagcloudVisualization.render(dummyTableGroup, vis.params, {
        resize: false,
        params: true,
        aggs: true,
        data: true,
        uiState: false,
      });

      const svgNode = domNode.querySelector('svg');
      expect(svgNode.outerHTML).toMatchSnapshot();
    });

    test('with resize', async function () {
      const tagcloudVisualization = new TagCloudVisualization(domNode, vis);
      await tagcloudVisualization.render(dummyTableGroup, vis.params, {
        resize: false,
        params: true,
        aggs: true,
        data: true,
        uiState: false,
      });

      domNode.style.width = '256px';
      domNode.style.height = '368px';
      await tagcloudVisualization.render(dummyTableGroup, vis.params, {
        resize: true,
        params: false,
        aggs: false,
        data: false,
        uiState: false,
      });

      const svgNode = domNode.querySelector('svg');
      expect(svgNode.outerHTML).toMatchSnapshot();
    });

    test('with param change', async function () {
      const tagcloudVisualization = new TagCloudVisualization(domNode, vis);
      await tagcloudVisualization.render(dummyTableGroup, vis.params, {
        resize: false,
        params: true,
        aggs: true,
        data: true,
        uiState: false,
      });

      setSVGElementGetBBox(256, 368);

      Object.defineProperties(window.SVGElement.prototype, {
        transform: {
          get: () => ({
            baseVal: {
              consolidate: () => {},
            },
          }),
          configurable: true,
        },
      });

      setHTMLElementOffset(256, 386);

      vis.params.orientation = 'right angled';
      vis.params.minFontSize = 70;
      await tagcloudVisualization.render(dummyTableGroup, vis.params, {
        resize: true,
        params: true,
        aggs: false,
        data: false,
        uiState: false,
      });

      const svgNode = domNode.querySelector('svg');
      expect(svgNode.outerHTML).toMatchSnapshot();
    });
  });

  function setupDOM(width, height) {
    domNode = document.createElement('div');
    setHTMLElementOffset(width, height);
    document.body.appendChild(domNode);
  }

  function teardownDOM() {
    domNode.innerHTML = '';
    document.body.removeChild(domNode);
  }
});
