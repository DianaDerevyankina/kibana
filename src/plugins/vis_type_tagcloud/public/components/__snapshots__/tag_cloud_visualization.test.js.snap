// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagCloudVisualizationTest TagCloudVisualization - basics simple draw 1`] = `"<svg width=\\"0\\" height=\\"0\\"><g width=\\"0\\" height=\\"0\\"><text style=\\"font-size: undefinedpx; font-style: normal; font-weight: normal; font-family: Open Sans, sans-serif; fill: #00a69b;\\" text-anchor=\\"middle\\" transform=\\"translate(0, 0)rotate(0)\\" data-test-subj=\\"CN\\">CN</text><text style=\\"font-size: undefinedpx; font-style: normal; font-weight: normal; font-family: Open Sans, sans-serif; fill: #00a69b;\\" text-anchor=\\"middle\\" transform=\\"translate(0, 0)rotate(0)\\" data-test-subj=\\"IN\\">IN</text><text style=\\"font-size: undefinedpx; font-style: normal; font-weight: normal; font-family: Open Sans, sans-serif; fill: #00a69b;\\" text-anchor=\\"middle\\" transform=\\"translate(0, 0)rotate(0)\\" data-test-subj=\\"US\\">US</text><text style=\\"font-size: undefinedpx; font-style: normal; font-weight: normal; font-family: Open Sans, sans-serif; fill: #00a69b;\\" text-anchor=\\"middle\\" transform=\\"translate(0, 0)rotate(0)\\" data-test-subj=\\"DE\\">DE</text><text style=\\"font-size: undefinedpx; font-style: normal; font-weight: normal; font-family: Open Sans, sans-serif; fill: #00a69b;\\" text-anchor=\\"middle\\" transform=\\"translate(0, 0)rotate(0)\\" data-test-subj=\\"BR\\">BR</text></g></svg>"`;

exports[`TagCloudVisualizationTest TagCloudVisualization - basics with param change 1`] = `"<svg width=\\"0\\" height=\\"0\\"><g width=\\"0\\" height=\\"0\\"><text style=\\"font-size: undefinedpx; font-style: normal; font-weight: normal; font-family: Open Sans, sans-serif; fill: #00a69b;\\" text-anchor=\\"middle\\" transform=\\"translate(0, 0)rotate(0)\\" data-test-subj=\\"CN\\">CN</text><text style=\\"font-size: undefinedpx; font-style: normal; font-weight: normal; font-family: Open Sans, sans-serif; fill: #00a69b;\\" text-anchor=\\"middle\\" transform=\\"translate(0, 0)rotate(0)\\" data-test-subj=\\"IN\\">IN</text><text style=\\"font-size: undefinedpx; font-style: normal; font-weight: normal; font-family: Open Sans, sans-serif; fill: #00a69b;\\" text-anchor=\\"middle\\" transform=\\"translate(0, 0)rotate(0)\\" data-test-subj=\\"US\\">US</text><text style=\\"font-size: undefinedpx; font-style: normal; font-weight: normal; font-family: Open Sans, sans-serif; fill: #00a69b;\\" text-anchor=\\"middle\\" transform=\\"translate(0, 0)rotate(0)\\" data-test-subj=\\"DE\\">DE</text><text style=\\"font-size: undefinedpx; font-style: normal; font-weight: normal; font-family: Open Sans, sans-serif; fill: #00a69b;\\" text-anchor=\\"middle\\" transform=\\"translate(0, 0)rotate(0)\\" data-test-subj=\\"BR\\">BR</text></g></svg>"`;

exports[`TagCloudVisualizationTest TagCloudVisualization - basics with resize 1`] = `"<svg width=\\"0\\" height=\\"0\\"><g width=\\"0\\" height=\\"0\\"><text style=\\"font-size: undefinedpx; font-style: normal; font-weight: normal; font-family: Open Sans, sans-serif; fill: #00a69b;\\" text-anchor=\\"middle\\" transform=\\"translate(0, 0)rotate(0)\\" data-test-subj=\\"CN\\">CN</text><text style=\\"font-size: undefinedpx; font-style: normal; font-weight: normal; font-family: Open Sans, sans-serif; fill: #00a69b;\\" text-anchor=\\"middle\\" transform=\\"translate(0, 0)rotate(0)\\" data-test-subj=\\"IN\\">IN</text><text style=\\"font-size: undefinedpx; font-style: normal; font-weight: normal; font-family: Open Sans, sans-serif; fill: #00a69b;\\" text-anchor=\\"middle\\" transform=\\"translate(0, 0)rotate(0)\\" data-test-subj=\\"US\\">US</text><text style=\\"font-size: undefinedpx; font-style: normal; font-weight: normal; font-family: Open Sans, sans-serif; fill: #00a69b;\\" text-anchor=\\"middle\\" transform=\\"translate(0, 0)rotate(0)\\" data-test-subj=\\"DE\\">DE</text><text style=\\"font-size: undefinedpx; font-style: normal; font-weight: normal; font-family: Open Sans, sans-serif; fill: #00a69b;\\" text-anchor=\\"middle\\" transform=\\"translate(0, 0)rotate(0)\\" data-test-subj=\\"BR\\">BR</text></g></svg>"`;
